src/cloudant/database.py

```diff
@@ -152,15 +152,12 @@ def create_document(self, data, throw_on_exists=False):
             corresponding to the new document in the database.
         """
         docid = data.get('_id', None)
-        doc = None
         if docid and docid.startswith('_design/'):
             doc = DesignDocument(self, docid)
         else:
             doc = Document(self, docid)
-        if throw_on_exists and doc.exists():
-            raise CloudantDatabaseException(409, docid)
         doc.update(data)
-        doc.create()
+        doc.create(throw_on_exists=throw_on_exists)
         super(CouchDatabase, self).__setitem__(doc['_id'], doc)
```

Comment:
I don't know that pushing the throw_on_exists functionality down to the Document.create is the right way to go.  For one thing throwing a `CloudantDatabaseException` from there does change current functionality somewhat in that if a user of the library has code currently written using `create` and is expecting an HTTPError then they would have to refactor their code due to this change.  Wouldn't it just be easier to do this here:

```
try:
    doc.create()
except HTTPError as error:
    if error.response.status_code == 409:
        if throw_on_exist:
            raise
    else:
        raise
```

That way the `throw_on_exist` remains where it was intended to be, in the database object.